README.md

```diff
@@ -322,6 +322,101 @@ export default {
 };
 ```
 
+## Built Engine Output
+
+### Eager Engines
+
+Eager engines are built approximately the same as existing addons. Differences
+are limited to consolidating the namespace of `app` code inside of an engine
+into the engine's namespace instead of the host application.
+
+Beyond that it adds in a configuration module for the engine, and nothing else. 
+It is a remarkably straightforward process.
+
+### Lazy Engines
+
+Lazy engines are built in the same way as eager engines, but their assets are
+not combined back into the host application's `vendor.js` file. This means that
+they are run through a separate and unique build process from what a default
+addon will go through, though it reaches out to the upstream implementation in
+Ember CLI where possible.
+
+A lazy engine's output (`lazy-engine`) looks like this:
+```
+dist
+├── assets
+│   ├── host-application.css
+│   ├── host-application.js
+│   ├── vendor.css
+│   └── vendor.js
+├── crossdomain.xml
+├── engines-dist
+│   └── lazy-engine
+│       ├── engine.js
```

Comment:
I think there was that and also @rwjblue's point about ensuring relative paths stay consistent (since things like `vendor.css` normally expect to be in `assets`).